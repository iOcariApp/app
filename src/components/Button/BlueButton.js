import React from "react";
import { colors } from "theme";

import Button from "./Button";

const BlueButton = props => <Button backgroundColor={colors.blue} {...props} />;

export default BlueButton;
