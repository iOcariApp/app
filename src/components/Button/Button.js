import React from "react";
import PropTypes from "prop-types";
import styles from "./Button.style";

import { Button } from "react-native-elements";

import { colors, button } from "theme";

const MyButton = ({ style, backgroundColor, ...rest }) => (
  <Button
    backgroundColor={backgroundColor}
    containerViewStyle={[styles.main, style]}
    buttonStyle={styles.button}
    borderRadius={button.radius}
    fontSize={button.fontSize}
    {...rest}
  />
);

MyButton.propTypes = {
  style: PropTypes.oneOfType([PropTypes.object, PropTypes.number]),
  backgroundColor: PropTypes.string,
};
MyButton.defaultProps = {
  style: {},
  backgroundColor: colors.mainColor,
};

export default MyButton;
