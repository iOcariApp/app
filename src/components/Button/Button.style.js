import { StyleSheet } from "react-native";
import { button } from "theme";

export default StyleSheet.create({
  main: {
    width: "100%",
    marginLeft: 0,
    marginRight: 0,
    marginTop: 0,
    marginBottom: 0,
  },
  button: {
    height: button.height,
  },
  smallButton: {
    height: button.smallHeight,
  },
});
