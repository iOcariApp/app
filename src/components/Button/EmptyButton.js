import React from "react";
import { PropTypes } from "prop-types";
import { StyleSheet } from "react-native";
import { button } from "theme";

import Button from "./Button";

const EmptyButton = ({ color, buttonStyle, ...rest }) => (
  <Button
    buttonStyle={StyleSheet.flatten([
      {
        borderColor: color,
        borderWidth: button.border,
      },
      buttonStyle,
    ])}
    color={color}
    backgroundColor="transparent"
    {...rest}
  />
);

EmptyButton.propTypes = {
  color: PropTypes.string,
  buttonStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.number]),
};
EmptyButton.defaultProps = {
  color: "white",
  buttonStyle: {},
};

export default EmptyButton;
