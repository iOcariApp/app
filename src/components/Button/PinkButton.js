import React from "react";
import { colors } from "theme";

import Button from "./Button";

const PinkButton = props => <Button backgroundColor={colors.pink} {...props} />;

export default PinkButton;
