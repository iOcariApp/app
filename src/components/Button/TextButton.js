import React from "react";
import { PropTypes } from "prop-types";

import Button from "./Button";

const TextButton = ({ color, ...rest }) => (
  <Button color={color} backgroundColor="transparent" {...rest} />
);

TextButton.propTypes = {
  color: PropTypes.string,
};
TextButton.defaultProps = {
  color: "white",
};

export default TextButton;
