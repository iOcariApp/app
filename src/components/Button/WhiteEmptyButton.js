import React from "react";
import { PropTypes } from "prop-types";
import { StyleSheet } from "react-native";
import { button } from "theme";

import Button from "./Button";

const WhiteEmptyButton = ({ buttonStyle, ...rest }) => (
  <Button
    buttonStyle={StyleSheet.flatten([
      buttonStyle,
      {
        borderColor: "white",
        borderWidth: 2,
        height: button.height - 2,
      },
    ])}
    backgroundColor="transparent"
    {...rest}
  />
);

WhiteEmptyButton.propTypes = {
  buttonStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.number]),
};
WhiteEmptyButton.defaultProps = {
  buttonStyle: {},
};

export default WhiteEmptyButton;
