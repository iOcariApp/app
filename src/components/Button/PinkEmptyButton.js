import React from "react";
import { colors } from "theme";

import EmptyButton from "components/Button/EmptyButton";

const PinkEmptyButton = props => <EmptyButton color={colors.pink} {...props} />;

export default PinkEmptyButton;
