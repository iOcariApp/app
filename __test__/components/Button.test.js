import React from "react";
import renderer from "react-test-renderer";

import Button from "components/Button";

test("renders correctly", () => {
  const tree = renderer.create(<Button />).toJSON();
  expect(tree).toMatchSnapshot();
});

describe("when receiving props", () => {
  it("should apply them correctly", () => {
    const tree = renderer
      .create(<Button style={{ width: 50 }} background="white" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
