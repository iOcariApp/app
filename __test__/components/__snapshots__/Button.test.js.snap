// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "marginLeft": 15,
        "marginRight": 15,
      },
      undefined,
      Array [
        Object {
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
          "width": "100%",
        },
        Object {},
      ],
      Object {
        "borderRadius": 4,
      },
    ]
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      pointerEvents="box-only"
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#9E9E9E",
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 19,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Object {
            "backgroundColor": "#F50057",
          },
          Object {
            "borderRadius": 4,
          },
          Object {
            "padding": 12,
          },
          undefined,
          undefined,
          undefined,
          Object {
            "height": 56,
          },
          undefined,
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Array [
              Object {
                "color": "white",
                "fontSize": 20,
              },
              undefined,
              Object {
                "fontSize": 17.5,
              },
              Object {
                "fontSize": 18,
              },
              undefined,
              undefined,
              undefined,
              undefined,
            ],
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`when receiving props should apply them correctly 1`] = `
<View
  style={
    Array [
      Object {
        "marginLeft": 15,
        "marginRight": 15,
      },
      undefined,
      Array [
        Object {
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
          "width": "100%",
        },
        Object {
          "width": 50,
        },
      ],
      Object {
        "borderRadius": 4,
      },
    ]
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      pointerEvents="box-only"
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#9E9E9E",
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 19,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Object {
            "backgroundColor": "#F50057",
          },
          Object {
            "borderRadius": 4,
          },
          Object {
            "padding": 12,
          },
          undefined,
          undefined,
          undefined,
          Object {
            "height": 56,
          },
          undefined,
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Array [
              Object {
                "color": "white",
                "fontSize": 20,
              },
              undefined,
              Object {
                "fontSize": 17.5,
              },
              Object {
                "fontSize": 18,
              },
              undefined,
              undefined,
              undefined,
              undefined,
            ],
          ]
        }
      />
    </View>
  </View>
</View>
`;
